superset-frontend/src/filters/components/Select/SelectFilterPlugin.tsx

```diff
@@ -109,6 +109,7 @@ export default function PluginFilterSelect(props: PluginFilterSelectProps) {
   } = formData;
   const groupby = ensureIsArray<string>(formData.groupby);
   const [col] = groupby;
+  const [initialColtypeMap] = useState(coltypeMap);
```

Comment:
I'm not sure if this proposal will be rellevant, but may be it can make sense instead of split state to different places we can control calls of `setDataMask` hook, for example - not call it if doesn't have filled own state or stuff like that, because if we will have some more logic for own state we can get same issue, or may be I wrong

Reply:
I think we should ultimately make the chart component check if a call to `buildQuery` actually returns a changed query. This way we could avoid retriggering queries when `ownState` changes but the actual `QueryObject`s or `QueryContext` doesn't change.